resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -97,4 +97,17 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
             }
         });
     }
+
+    /**
+     * @param anyInt any integer value to covert
+     * @param len max limit of range
+     * @return integer in (0, len] range, using formula {@code (anyInt % len + anyInt) % len}
+     */
+    protected static int toIndex(int anyInt, int len) {
+        int index = anyInt % len;
+        if (index < 0) {
+            index = index + len;
```

Comment:
I thought I was crazy but execute the following and see the base of your formula is always 1 and reaches length:
```
    for (int i = -10; i < 0; i++) {
      final int index = i % 5 + 5; // +5 = length
      System.out.println("Without -1 for i = " + i + ", index = " + index);
      System.out.println("With -1 for i = " + i + ", index = " + index);
    }
```
Output:
```
Without -1 for i = -10, index = 5
With -1 for i = -10, index = 4
Without -1 for i = -9, index = 1
With -1 for i = -9, index = 0
Without -1 for i = -8, index = 2
With -1 for i = -8, index = 1
Without -1 for i = -7, index = 3
With -1 for i = -7, index = 2
Without -1 for i = -6, index = 4
With -1 for i = -6, index = 3
Without -1 for i = -5, index = 5
With -1 for i = -5, index = 4
Without -1 for i = -4, index = 1
With -1 for i = -4, index = 0
Without -1 for i = -3, index = 2
With -1 for i = -3, index = 1
Without -1 for i = -2, index = 3
With -1 for i = -2, index = 2
Without -1 for i = -1, index = 4
With -1 for i = -1, index = 3
```